Content.Shared/_RMC14/FarSight/FarSightSystem.cs

```diff
@@ -42,10 +48,47 @@ private void OnFarSightAction(Entity<FarSightItemComponent> ent, ref FarSightAct
         }
         else
         {
+            if (TryComp<EyeComponent>(user, out var eye))
+                _eye.SetMaxZoom(user, eye.Zoom);
+
             _eye.ResetZoom(user);
         }
 
         _actions.SetToggled(ent.Comp.Action, ent.Comp.Enabled);
         _appearance.SetData(ent, FarSightItemVisuals.Active, ent.Comp.Enabled);
     }
+
+    private void OnFarSightEquipped(Entity<FarSightItemComponent> ent, ref GotEquippedEvent args)
+    {
+        var user = args.Equipee;
+
+        if (!_inventory.InSlotWithFlags((ent, null, null), ent.Comp.Slots))
+            return;
+
+        if (ent.Comp.Enabled)
+        {
+            _eye.SetMaxZoom(user, ent.Comp.Zoom);
+            _eye.SetZoom(user, ent.Comp.Zoom);
+        }
+        else
+        {
+            if (TryComp<EyeComponent>(user, out var eye))
+                _eye.SetMaxZoom(user, eye.Zoom);
+
+            _eye.ResetZoom(user);
+        }
```

Comment:
This should be its own method so that this and OnFarSightAction don't have to repeat it